src/backend/distributed/connection/connection_management.c

```diff
@@ -738,7 +739,7 @@ FinishConnectionListEstablishment(List *multiConnectionList)
 
 				CHECK_FOR_INTERRUPTS();
 
-				if (InterruptHoldoffCount > 0 && (QueryCancelPending || ProcDiePending))
+				if (IsCancellationReceived())
```

Comment:
This catches a specific situation when we are in a HOLD_INTERRUPTS block to avoid stack unwinding  / errors. I wonder if that should be reflected in the name. e.g IsHoldOffCancellationReceived

Reply:
@marcocitus I have also updated the comment for the method to be more clear why we have that.